backend-shared/config/db/clustercredentials_test.go

```diff
@@ -0,0 +1,78 @@
+package db
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestCreateandDeleteClusterCredentials(t *testing.T) {
+	ctx := context.Background()
+	dbq, err := NewUnsafePostgresDBQueries(true, true)
+	if !assert.NoError(t, err) {
+		return
+	}
+	defer dbq.CloseDatabase()
+	clusterCreds := ClusterCredentials{
+		Host:                        "test-host",
+		Kube_config:                 "test-kube_config",
+		Kube_config_context:         "test-kube_config_context",
+		Serviceaccount_bearer_token: "test-serviceaccount_bearer_token",
+		Serviceaccount_ns:           "test-serviceaccount_ns",
+	}
+	err = dbq.CreateClusterCredentials(ctx, &clusterCreds)
+	if !assert.NoError(t, err) {
+		return
+	}
+	fetchedCluster := ClusterCredentials{
+		Clustercredentials_cred_id: clusterCreds.Clustercredentials_cred_id,
+	}
+	err = dbq.GetClusterCredentialsById(ctx, &fetchedCluster)
+	assert.NoError(t, err)
+	assert.Equal(t, clusterCreds, fetchedCluster)
+
+	count, err := dbq.DeleteClusterCredentialsById(ctx, clusterCreds.Clustercredentials_cred_id)
+	if !assert.NoError(t, err) {
+		return
+	}
+	assert.Equal(t, 1, count)
+	err = dbq.GetClusterCredentialsById(ctx, &fetchedCluster)
+	assert.Equal(t, err, NewResultNotFoundError("UnsafeGetClusterCredentialsById"))
```

Comment:
We can instead use `assert.True(t, IsResultNotFoundError(err))` here, rather than creating a new error and comparing them as above.